### ISSUE TITLE
Gam, incorrectly wont add 11th delegate

### ISSUE THREAD

**Issue by clader** [jay0lee/GAM#1180]
I was trying to add a number of delegates to a users inbox via gam.  I was doing this one at a time and all was working just fine, until I got to the 11th one and got this error:

```
gam user john.doe@acme.com delegate to jane.doe@acme.com
Giving jane.doe@acme.com delegate access to john.doe@acme.com (1/1)

ERROR: 400: Delegator user cannot have any more delegates - failedPrecondition
```

This is confusing since the maximum number of delegates for a Gsuite enterprise domain is supposed to be 25 (and I know we have added more than 10 to a single user in the past)
https://developers.google.com/admin-sdk/email-settings/limits.

I'm using gam v5.0.4

**Comment by jay0lee** [comment]
What does:

gam user john.die@acme.com show delegates

Show interns of number of existing delegates?

**Comment by clader** [comment]
running: 
```
gam user john.doe@acme.com show delegate
```
Returns a list of 10 accepted delegates for the user. (don't want to paste the results here for privacy reasons).

Also I know of other accounts that have more than 10 delegates and doing a show delegates on them shows >10 delegates (but less than 25).  And when trying to use gam to add a new delegate to these accounts that have more than 10 delegates returns the same error.

**Comment by taers232c** [comment]
I can go higher than 10 on a GAFE domain.

Ross

**Comment by jay0lee** [comment]
What does

gam info user john.doe@acme.com

show? Feel free to anonymize the data but I'm particularly interested to
know the license type of the user.

Also what happens if you log into the user and try to add an 11th from the
Gmail UI?

By the way, probably worth opening a support case with Google about the
error with Google.  Especially if it reproduces in the Gmail UI. GAM does
not enforce these limits, it only reports them when Google servers tell it
the limit was hit and it has no control over whether that's 10, 25 or 1,000.


Jay

**Comment by clader** [comment]
I was able to successfully invite an 11th user by logging into the gsuite user and adding them through gmail settings->acounts.

I was also able to add a 12th user using g.bettercloud.com, which (I believe) is using the same api's that gam does.

Here is the anonymized gam output you requested:
```
> gam info user john.doe@acme.com

**Comment by jay0lee** [comment]
Does the issue still reproduce for that user after you added one via the UI?

**Comment by clader** [comment]
Yeah, actually I was able to add a 13th and 14th delegate after I added in the UI.
But I don't think adding it in the UI is what caused it to start working.

There were actually 2 problems when I added the initial 11 delegates via gam:
1) the first 10 people I added as delegate were never able to actually access the inbox (it showed in their account drop down as a delegated inbox, but when they clicked it they got a 403 Forbidden error)
2) when trying to add the 11th, I got the error reported above.

It wasn't until like an hour later that I tried to add the  11th/12th delegates via the UI and Bettercloud, and the original 10 delegates I added from gam still weren't working, so I deleted all those 10 delegates via the UI, and then re-added them via gam and it immediately started working for those 10 users, and then I could add more (13th, 14th) via gam.

So I'm wondering if this might have actually been a licensing issue, in that when I first created the account in gsuite, it didn't get auto-assigned the enterprise license right away.

So, the first 10 that I added via gam, were added when the account didn't have a full enterprise license allocated, and wouldn't work.  And then after like an hour the full license got added, but those first 10 were still tainted somehow, and only once I removed and re-added (after the enterprise license was granted) did everything start working normally.

I don't know how to prove this delayed license allocation is what caused this, but it's possible.  So I guess you can mark this closed for now.

**Comment by aston-wooller** [comment]
Just adding a comment here because it's the first result when I searched for the error GAM was presenting. In the case where I received this error, I had to wait 2 days before it let me add the rest of the delegates. 

It didn't appear to be license related, my problem account had a license assigned and any license should allow the addition of up to 1000 delegates with the current delegate limits.

In my case, adding a user through the UI sent the confirmation request email, but after clicking on the confirm link it presented another error which I unfortunately didn't screenshot and it refused to confirm the addition.

Anyway, long story short is you've gotta wait 24-48 hours and it will eventually let you add delegates beyond the first 10.
